tests/test_github_interrupted_sync_remove_stream.py

```diff
@@ -0,0 +1,204 @@
+                                
+                            if stream in interrupted_repo_bookmark.keys():
+                                interrupted_bookmark = self.dt_to_ts(interrupted_repo_bookmark[stream]["since"], self.BOOKMARK_FORMAT)
+                            
+                                if stream != removed_stream:
+                                    final_sync_stream_bookmark = self.dt_to_ts(final_bookmark.get(stream, {}).get("since"), self.BOOKMARK_FORMAT)
+
+                                    # Verify state ends with the same value for common streams after both full and interrupted syncs
+                                    self.assertEqual(full_sync_stream_bookmark, final_sync_stream_bookmark)
+
+                                    # Verify resuming sync only replicates records with replication key values greater or equal to
+                                    # the interrupted_state for streams that completed were replicated during the interrupted sync.
+                                    for record in interrupted_records:
+                                        with self.subTest(record_primary_key=record[expected_primary_keys[0]]):
+                                            rec_time = self.dt_to_ts(record[expected_replication_key], self.RECORD_REPLICATION_KEY_FORMAT)
+                                            self.assertGreaterEqual(rec_time, interrupted_bookmark)
+
+                                            # Verify all interrupted recs are in full recs
+                                            self.assertIn(record, full_records,  msg='Incremental table record in interrupted sync not found in full sync')
+
+                                    # Record count for all streams of interrupted sync match expectations
+                                    full_records_after_interrupted_bookmark = 0
+                                    for record in full_records:
+                                        rec_time = self.dt_to_ts(record[expected_replication_key], self.RECORD_REPLICATION_KEY_FORMAT)
+                                        self.assertGreater(rec_time, start_date, msg=f"{expected_replication_key} {stream} {repository} {record}")
+
+                                        if (rec_time >= interrupted_bookmark):
+                                            full_records_after_interrupted_bookmark += 1
+
+                                    self.assertEqual(full_records_after_interrupted_bookmark, interrupted_record_count, \
+                                                        msg="Expected {} records in each sync".format(full_records_after_interrupted_bookmark))
+                                else:
+                                    # Verify the bookmark has not advanced for the removed stream
+                                    self.assertEqual(final_sync_stream_bookmark, full_sync_stream_bookmark)
```

Comment:
I see **_final_sync_stream_bookmark_** is being declared inside the if cond at line 157. So, it would have the bookmark value for stream from previous iteration. So, I guess this assertion is wrong(I might be wrong). and this integation test is failing in my dev-vm with below error
`AssertionError: 1659092567 != 1658150542`

Reply:
Yes, you are right, I completely missed that. Updated the test to declare the var outside the if block.

Reply:
circleCI is failing but approving this since rate limit exception is being handled in different branch